src/storage/src/hummock/compactor.rs

```diff
@@ -294,6 +294,7 @@ impl Compactor {
     pub async fn compact_shared_buffer(
         context: Arc<CompactorContext>,
         payload: UploadTaskPayload,
+        water_epoch: HummockEpoch,
```

Comment:
can we use `watermark_epoch` to replace `water_epoch` ? 
`water_epoch` is a little confuse to me at the first time